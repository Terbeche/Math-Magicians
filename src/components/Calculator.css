#calculator-container{
  display: flex;
  gap: 2rem;
  justify-content: space-around;
  margin-top: 5rem;
}
#calculator {
  display: grid;
  grid-template: 1fr 1fr 1fr 1fr 1fr 1fr/1fr 1fr 1fr 1fr;
  height: 20rem;
  width: 60%;
}

#result {
  grid-column: span 4;
  border: solid black 0.1rem;
  margin: 0;
  background-color: #aaa;
  text-align: right;
  font-size: 2rem;
  padding: 0.2rem;
}

#zero-button {
  grid-column: span 2;
}

.operator {
  background-color: #d1802a;
}

button:hover {
  cursor: pointer;
}

.calculator-title {
  color:saddlebrown;
}