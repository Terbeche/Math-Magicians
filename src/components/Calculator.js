/* eslint-disable jsx-a11y/label-has-associated-control */
import { useState, React } from 'react';
import './Calculator.css';
import calculate from '../logic/calculate';
// import operate from '../logic/operate';

const Calculator = () => {
  const [state, setState] = useState({ total: 0, next: 0 });
  // const { total, next } = this.state;

  const handleClick = (e) => {
    const operation = e.target.value;
    const result = calculate(state, operation);
    setState(result);
  };

  const handleTextChange = (e) => {
    setState({
      total: e.target.value,
    });
  };

  return (
    <div id="calculator">
      <p id="result" onChange={handleTextChange}>{state.next || state.total || 0}</p>
      <button type="button" onClick={handleClick} value="AC">AC</button>
      <button type="button" onClick={handleClick} value="+/-">+/-</button>
      <button type="button" onClick={handleClick} value="%">%</button>
      <button className="operator" type="button" onClick={handleClick} value="รท">/</button>
      <button type="button" onClick={handleClick} value="7">7</button>
      <button type="button" onClick={handleClick} value="8">8</button>
      <button type="button" onClick={handleClick} value="9">9</button>
      <button className="operator" type="button" onClick={handleClick} value="x">x</button>
      <button type="button" onClick={handleClick} value="4">4</button>
      <button type="button" onClick={handleClick} value="5">5</button>
      <button type="button" onClick={handleClick} value="6">6</button>
      <button className="operator" type="button" onClick={handleClick} value="-">-</button>
      <button type="button" onClick={handleClick} value="1">1</button>
      <button type="button" onClick={handleClick} value="2">2</button>
      <button type="button" onClick={handleClick} value="3">3</button>
      <button className="operator" type="button" onClick={handleClick} value="+">+</button>
      <button id="zero-button" type="button" onClick={handleClick} value="0">0</button>
      <button type="button" onClick={handleClick} value=".">.</button>
      <button className="operator" type="button" onClick={handleClick} value="=">=</button>
    </div>
  );
};

export default Calculator;
