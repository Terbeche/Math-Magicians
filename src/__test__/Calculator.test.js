import React from 'react';
import renderer from 'react-test-renderer';
import { BrowserRouter } from 'react-router-dom';
import Calculator from '../components/Calculator';

describe('Testing the Home', () => {
  it('Calculator is rendered', () => {
    const calculator = renderer
      .create(<BrowserRouter><Calculator /></BrowserRouter>)
      .toJSON();
    expect(calculator).toMatchSnapshot();
  });
});
