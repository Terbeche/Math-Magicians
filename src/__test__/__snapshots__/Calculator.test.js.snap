// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Testing the Home Calculator is rendered 1`] = `
<div
  id="calculator-container"
>
  <h2
    className="calculator-title"
  >
    Let's do some math!
  </h2>
  <div
    id="calculator"
  >
    <p
      id="result"
      onChange={[Function]}
    >
      0
    </p>
    <button
      onClick={[Function]}
      type="button"
      value="AC"
    >
      AC
    </button>
    <button
      onClick={[Function]}
      type="button"
      value="+/-"
    >
      +/-
    </button>
    <button
      onClick={[Function]}
      type="button"
      value="%"
    >
      %
    </button>
    <button
      className="operator"
      onClick={[Function]}
      type="button"
      value="รท"
    >
      /
    </button>
    <button
      onClick={[Function]}
      type="button"
      value="7"
    >
      7
    </button>
    <button
      onClick={[Function]}
      type="button"
      value="8"
    >
      8
    </button>
    <button
      onClick={[Function]}
      type="button"
      value="9"
    >
      9
    </button>
    <button
      className="operator"
      onClick={[Function]}
      type="button"
      value="x"
    >
      x
    </button>
    <button
      onClick={[Function]}
      type="button"
      value="4"
    >
      4
    </button>
    <button
      onClick={[Function]}
      type="button"
      value="5"
    >
      5
    </button>
    <button
      onClick={[Function]}
      type="button"
      value="6"
    >
      6
    </button>
    <button
      className="operator"
      onClick={[Function]}
      type="button"
      value="-"
    >
      -
    </button>
    <button
      onClick={[Function]}
      type="button"
      value="1"
    >
      1
    </button>
    <button
      onClick={[Function]}
      type="button"
      value="2"
    >
      2
    </button>
    <button
      onClick={[Function]}
      type="button"
      value="3"
    >
      3
    </button>
    <button
      className="operator"
      onClick={[Function]}
      type="button"
      value="+"
    >
      +
    </button>
    <button
      id="zero-button"
      onClick={[Function]}
      type="button"
      value="0"
    >
      0
    </button>
    <button
      onClick={[Function]}
      type="button"
      value="."
    >
      .
    </button>
    <button
      className="operator"
      onClick={[Function]}
      type="button"
      value="="
    >
      =
    </button>
  </div>
</div>
`;
